terminal.py

```diff
@@ -308,5 +305,93 @@ def terminal():
                 print_goodbye()
 
 
+def install():
+    if os.path.isfile("data.txt"):
+        with open("data.txt") as f:
+            json_data = json.load(f)
+        new = json_data["installed"]
+
+    else:
+        with open("data.txt", "w+") as f:
+            data = {"installed": False}
+            json.dump(data, f)
+        new = False
+
+    if not new:
+        if platform.system() == "Darwin":
+            # Installation for Mac M1
+            if (
+                subprocess.check_output(
+                    ["sysctl", "-n", "machdep.cpu.brand_string"]
+                ).decode("utf-8")
+                == "Apple M1\n"
+            ):
+                if (
+                    input("Would you like to install Gamestonk Terminal?Y/N").lower()
+                    == "y"
+                ):
+                    subprocess.run(
+                        [
+                            "arch",
+                            "-x86_64",
+                            "/usr/bin/python3",
+                            "-m",
+                            "pip",
+                            "install",
+                            "--user",
+                            "--upgrade",
+                            "virtualenv",
+                        ],
+                        check=True,
```

Comment:
Logic like that will be very difficult to implement, test and maintain.

Ignoring testing and maintenance for a moment, how are you handling failures coming from the subprocess.run?